### PR TITLE
Upgrade dependencies

### LINKED ISSUE #109: Working with the unified `dart` tool
Starting from dart 2.10,  a new unified `dart` command (consistent with `flutter`) has been added. According to the [announcement](https://medium.com/dartlang/announcing-dart-2-10-350823952bd5#837b:~:text=Next%20year%20we%20expect%20to%20ship,tool%20when%20running%20Dart%20commands%20now), by next year other smaller tools will be removed. 

Here are the changes we need to do -
- [x] Use `dart format` instead of `dartfmt` for [formatting generated files](https://github.com/dart-lang/ffigen/blob/master/lib/src/code_generator/library.dart#L99).
- [x] Update readme(s) and comments to use `dart pub get` instead of `pub get`.

We should wait for these tools to be removed (as long as we can) before we update `ffigen`'s SDK constraint to be >=2.10,
OR, we can also try calling `dart format` and if that fails, try running `dartfmt`.

cc @dcharkes

### LINKED ISSUE #128: Unsound Null-Safety
We currently do not support sound null safety (added by #122).

Tasks Remaining: 
- [x] Migrate all dependencies
  - [x] Args
  - [x] logging
  - [x] cli_util
  - [x] glob
  - [x] pub_semver
- [x] Remove all `--no-sound-null-safety` flags.

### PR DESCRIPTION
**Pull request by mannprerak2**
Closes #128, Closes #109 
- Updated all dependencies to null safe versions.
- Cleanup: removed `dartfmt` usage since supported SDK versions are above 2.10.
- Update changelog and version (2.0.0, preview release)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# 2.0.0
+- Upgraded all dependencies. `package:ffigen` now runs with sound null safety.
```
Comment (dcharkes) [inline CHANGELOG.md:2]: 🎉

File: lib/src/header_parser/utils.dart [hunk 2]
```diff
@@ -100,19 +100,19 @@ extension CXCursorExt on clang_types.CXCursor {
 
   String sourceFileName() {
     final cxsource = clang.clang_getCursorLocation(this);
-    final cxfilePtr = allocate<Pointer<Void>>();
-    final line = allocate<Uint32>();
-    final column = allocate<Uint32>();
-    final offset = allocate<Uint32>();
+    final cxfilePtr = malloc<Pointer<Void>>();
```
Comment (dcharkes) [inline lib/src/header_parser/utils.dart:103]: You might want to use `calloc` instead. It 0-initializes memory.

File: pubspec.yaml [hunk 3]
```diff
@@ -3,24 +3,24 @@
...
-  quiver: ^3.0.0-nullsafety.2
-  args: ^1.6.0        # Not available
-  logging: ^0.11.4    # test->coverage->logging
-  cli_util: ^0.2.0    # test->analyzer->cli_util
-  glob: ^1.0.3        # test->analyzer->glob
-  pub_semver: ^1.4.4  # test->analyzer->pub_semver
+  ffi: ^1.0.0
+  yaml: ^3.0.0
+  path: ^1.8.0
+  quiver: ^3.0.0-nullsafety.3
```
Comment (dcharkes) [inline pubspec.yaml:17]: lgtm, we can always do  1.0.1 release if we want to clean this up.

File: .github/workflows/test-package.yml [hunk 4]
```diff
@@ -44,15 +44,15 @@ jobs:
       - uses: actions/checkout@v2
       - uses: dart-lang/setup-dart@v0.1
         with:
-          channel: dev # TODO(127): Revert to stable.
+          channel: beta # TODO(127): Revert to stable.
```
[the remark this answers is not in the document]
Comment (dcharkes) [reply .github/workflows/test-package.yml:47]: Maybe update the comment to reflect this.

**Review by dcharkes — APPROVED** [review]

**Comment by mannprerak2** [comment]
Done, @dcharkes, you can merge this and create a preview release now 🎊

File: lib/src/header_parser/utils.dart [hunk 5]
```diff
@@ -52,7 +52,7 @@ void logTuDiagnostics(
 
 extension CXSourceRangeExt on Pointer<clang_types.CXSourceRange> {
   void dispose() {
-    free(this);
+    malloc.free(this);
```
Comment (dcharkes) [inline lib/src/header_parser/utils.dart:55]: more malloc

File: lib/src/header_parser/utils.dart [hunk 6]
```diff
@@ -275,9 +275,9 @@ extension DynamicCStringArray on Pointer<Pointer<Utf8>> {
   // Properly disposes a Pointer<Pointer<Utf8>, ensure that sure length is correct.
   void dispose(int length) {
     for (var i = 0; i < length; i++) {
-      free(this[i]);
+      malloc.free(this[i]);
```
Comment (dcharkes) [inline lib/src/header_parser/utils.dart:278]: more malloc

File: lib/src/header_parser/utils.dart [hunk 7]
```diff
@@ -275,9 +275,9 @@ extension DynamicCStringArray on Pointer<Pointer<Utf8>> {
...
     }
-    free(this);
+    malloc.free(this);
```
Comment (dcharkes) [inline lib/src/header_parser/utils.dart:280]: more malloc

File: lib/src/header_parser/utils.dart (hunk 5 above)
Comment (mannprerak2) [reply lib/src/header_parser/utils.dart]: Done, replaced all `malloc` with `calloc`.
Is there any difference in using `malloc` vs `calloc` when freeing?

Comment (dcharkes) [reply lib/src/header_parser/utils.dart]: No, both just use normal `free`. But it's nice to be consistent.

**Pull request merged**